### PR TITLE
[PERFSCALE-2994]:  Updates to SDN-->OVNK Script

### PR DESCRIPTION
**Pull request by krishvoor**
## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Documentation Update

## Description

Updates the existing sdn2ovnk scripts, capture/index time taken for respective steps

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Review by mukrishn — APPROVED** [review]

**Comment by svetsa-rh** [comment]
@krishvoor Can we rename the workload to reflect that it is for self-managed and offline migration? 
Once we have scripts for live migration, we will end up in name conflict situation. Proactively naming appropriately might help reduce rework. WDYT?

**Comment by svetsa-rh** [comment]
@krishvoor @mukrishn - Few clarification questions:
1) Is this script only for self-managed clusters? I see the initial commit stating "Managed clusters" but the update in this PR states "Self-managed". Confused :( 

2) Also, we have Live migration (testing in progress by me) and Offline migration (testing in progress by Harsha) - to avoid duplication, should we have same workload handle both or do we want to have them separate and hence, name accordingly?

3) Is the intention of this workload to be be able to test in Prow easily or both within and outside of Prow or outside prow only?

Soon, I will have to start adding automation tests for SDN to OVN Live migration for managed services and we also have same for self-managed services, so trying to get some clarification.

TIA!

**Comment by krishvoor** [comment]
> @krishvoor @mukrishn - Few clarification questions:
> 
> 1. Is this script only for self-managed clusters? I see the initial commit stating "Managed clusters" but the update in this PR states "Self-managed". Confused :(

This script is tested and validated only on the Self-Managed clusters, more details about the testing can be found here:
https://docs.google.com/document/d/1JULTxrZN9jCTXp9WSWj3j8vp3vNe0smbXSs56CHPJ_U/edit?usp=sharing 
Please ignore the individual commit messages :-) I have added JIRA details to the GH issue title

> 2. Also, we have Live migration (testing in progress by me) and Offline migration (testing in progress by Harsha) - to avoid duplication, should we have the same workload handle both or do we want to have them separate and hence, name accordingly?

FYI - Offline migration testing on the Self-Managed cluster(s) is now completed.
Feel free to repurpose the script to add new functionalities like live migration, if it's too much of a rework we can tackle it separately.  

> 3. Is the intention of this workload to be be able to test in Prow easily or both within and outside of Prow or outside prow only?

The PR intends to add the delta changes found during Offline CNI Testing.
It can be easily integrated into Prow, afaik no plans to run Offline CNI migration in Prow.

> Soon, I will have to start adding automation tests for SDN to OVN Live migration for managed services and we also have same for self-managed services, so trying to get some clarification.
> 
> TIA!

Let me know if you have questions - happy to answer them

**Comment by vishnuchalla** [comment]
@krishvoor looks like it just needs signed commits.

File: workloads/sdn2ovn/run.sh [hunk 1]
```diff
@@ -31,51 +37,115 @@ _check_pods(){
...
+    return 0
+}
+
 echo "Migration of cluster default CNI to OVNKubernetes"
 
-N_TYPE=$(oc get Network.operator.openshift.io cluster  -o json | jq -r '.spec.defaultNetwork.type')
+echo "Take back up of cluster network configuration"
+oc get Network.config.openshift.io cluster -o yaml > $TEMP_DIR/cluster-openshift-sdn.yaml
+BEFORE_N_TYPE=$(oc get Network.operator.openshift.io cluster  -o json | jq -r '.spec.defaultNetwork.type')
 echo "Current CNI is $N_TYPE"
```
Comment (venkataanil) [inline workloads/sdn2ovn/run.sh:68]: Typo here. N_TYPE should have been replaced with BEFORE_N_TYPE

**Review by venkataanil — APPROVED** [review]
LGTM

File: workloads/sdn2ovn/run.sh (hunk 1 above)
Comment (krishvoor) [reply workloads/sdn2ovn/run.sh:68]: I have updated - ptal!

**Pull request merged**